### PR TITLE
feat: add volumes (PVC) to kubernetes

### LINKED ISSUE #7341: Add "Volumes" section for Kubernetes page.
### Is your feature request related to a problem? Please describe

See: https://github.com/containers/podman-desktop/issues/7190

We should be adding PVC (Persistent Volume Claims), NOT Volumes.

The reasoning is that Volumes = administrative (adding nfs, local storage, etc.).

Persistent Volume Claims = What the Pod / Deployment / etc is actually requesting and most likely used to confirm volume claim when developing.

### Describe the solution you'd like

Added to Kubernetes dashboard.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
feat: add volumes (PVC) to kubernetes

### What does this PR do?

* Adds PVC to Kubernetes
* Lists in the dashboard, available to edit, delete, etc.

### Screenshot / video of UI

Screenshot 2024-06-13 at 9 18 00 AM

https://github.com/containers/podman-desktop/assets/6422176/a438eda6-b41e-46ab-85b2-92f35ffc025e

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/7341

### How to test this PR?

- [X] Tests are covering the bug fix or the new feature

Deploy a Kubernetes artifact with a PVC and view / check / play around
with the new dashboard for PVC.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by benoitf** [comment]
should PVC be a subfolder of kube ?

**Comment by cdrage** [comment]
> should PVC be a subfolder of kube ?

EDIT: Nope, it is in the right place like Deployments, Node, Ingress-Routes, etc.

**Comment by cdrage** [comment]
I have also created https://github.com/containers/podman-desktop/issues/7641 with regards to a new icon as right now it is using the podman volumes icon. FYI @ekidneyrh

**Review by lstocchi — APPROVED** [review]
Played with it and all pvcs i created were displayed/deleted correctly. LGTM

File: packages/renderer/src/lib/pvc/PVCList.svelte [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      <span>On {selectedItemsNumber} selected items.</span>
+    {/if}
+    <div class="flex grow justify-end">
+      <KubernetesCurrentContextConnectionBadge />
+    </div>
+  </svelte:fragment>
+
+  <div class="flex min-w-full h-full" slot="content">
+    <Table
+      kind="pvc"
```
Comment (deboer-tim) [inline packages/renderer/src/lib/pvc/PVCList.svelte]: Should be capitalized? i.e. should be 'No PVCs', 'Delete PVC', vs current 'No pvcs'.

File: packages/renderer/src/lib/pvc/PVCColumnMode.svelte [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import type { PVCUI } from './PVCUI';
+
+export let object: PVCUI;
+
+// Determine the icon and color based on the access mode, with comments explaining each icon choice
+function getModeAttributes(mode: string) {
+  switch (mode) {
+    case 'ReadWriteOnce':
+      // faLock: Represents exclusive write access (one writer), indicative of the ReadWriteOnce mode
+      return { color: 'text-green-600', icon: faLock };
```
Comment (deboer-tim) [inline packages/renderer/src/lib/pvc/PVCColumnMode.svelte]: These should really be pulling from the color registry now. I guess we can defer, but maybe open a follow-up issue to create color vars for all of the Kube objects?

File: packages/renderer/src/lib/pvc/PVCEmptyScreen.svelte [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<script lang="ts">
+import { EmptyScreen } from '@podman-desktop/ui-svelte';
+
+import PVCIcon from '../images/PVCIcon.svelte';
+</script>
+
+<EmptyScreen icon="{PVCIcon}" title="No pvcs" message="Try switching to a different context or namespace" />
```
Comment (deboer-tim) [inline packages/renderer/src/lib/pvc/PVCEmptyScreen.svelte]: Capitalize 'No PVCs'?

**Review by axel7083 — COMMENTED** [review]
Is the green a bit weird ?

| Node | PVC |
| --- | --- |
| image | image |

It might be because it is pending

image

But we should not be in some warning/orange/yellow state for pending ? 

image

**Review by deboer-tim — APPROVED** [review]
Would have preferred to see this as 3+ PRs, getting on the large side to review and test. Everything seems to be there though and worked well in testing. 🚀 Just some minor comments.

**Comment by cdrage** [comment]
> Is the green a bit weird ?
> 
> Node	PVC
> image	image
> It might be because it is pending
> 
> image
> 
> But we should not be in some warning/orange/yellow state for pending ?
> 
> image

For this, I use 'STARTING' which by default uses that darker green, since Starting = Pending as it's waiting for it.

I opened up issue: https://github.com/containers/podman-desktop/issues/7676

**Comment by cdrage** [comment]
@axel7083 @deboer-tim @lstocchi Updated the PR with the above changes and ready for another review!

Next PR with k8s features I'll split it up into two (one for details, other for just listing..) so it's much easier.

**Comment by lstocchi** [comment]
It would be good to add a new commit next time so one just gives a quick look at that. Mainly for big PR like this. I guess you just updated the color and texts pointed by Tim. So, LGTM!!

**Review by axel7083 — APPROVED** [review]
Great job, works well 🚀

**Comment by cdrage** [comment]
Going to set this to auto-merge this so we can work on getting the icons in next and any subsequent PR's. Thanks everyone for the reviews!

**Pull request merged**